core/src/main/scala/org/apache/spark/serializer/SerializerManager.scala

```diff
@@ -143,7 +179,25 @@ private[spark] class SerializerManager(defaultSerializer: Serializer, conf: Spar
     val stream = new BufferedInputStream(inputStream)
     getSerializer(implicitly[ClassTag[T]])
       .newInstance()
-      .deserializeStream(wrapForCompression(blockId, stream))
+      .deserializeStream(wrapStream(blockId, stream))
       .asIterator.asInstanceOf[Iterator[T]]
   }
 }
+
+private[spark] object SerializerManager {
+  /**
+   * Setup the cryptographic key used by IO encryption in credentials. The key is generated using
+   * [[KeyGenerator]]. The algorithm and key length is specified by the [[SparkConf]].
+   */
+  def initShuffleEncryptionKey(conf: SparkConf, credentials: Credentials): Unit = {
```

Comment:
I still think this makes more sense in `SecurityManager`; also, this is more than shuffle, so maybe just `initEncryptionKey` or ``initIOEncryptionKey`.